synadm/cli/user.py

```diff
@@ -500,6 +500,80 @@ def modify(ctx, helper, user_id, password, password_prompt, display_name,
         click.echo("Abort.")
 
 
+# for these placeholders, function similarly to the modify command, but in
+# it's own command.
+# TODO: find 3pid modify command
+
+
+@user.command()
+@click.argument("user_id", type=str)
+@click.argument("mxc_uri", type=str)
+@click.pass_obj
+def profile_picture(helper, user_id, mxc_uri):
+    """
+    Set a profile picture for a user by the MXC URI.
+    """
+    # TODO
+    pass
+
+
+@user.command()
+@click.argument("user_id", type=str)
+@click.argument("display_name", type=str)
+@click.pass_obj
+def display_name(helper, user_id, display_name):
+    """
+    Set the display name of a user.
+    """
+    # TODO
+    pass
+
+
+@user.command()
+@click.argument("user_id", type=str)
+@click.pass_obj
+def reactivate(helper, user_id):
+    # TODO
+    # TODO: does this require a password? merge into password? user
```

Comment:
It definitely does require a password.

Look for this phrase "Note: the password field must also be set if both of the following are true:" around here: https://matrix-org.github.io/synapse/latest/admin_api/user_admin_api.html#create-or-modify-account

Reply:
I basically agree but on the other hand: why not borrow the ideas we have over at the `user password` command?

I think we have both: stdin and cli arg

I still think that securitywise it's an admin's choice whether they want to eg. modify 10 testusers on a playground server - scripted! Or they'd like to securely modify s real user's account on the cli.

It's an old unix philosophy: Know what you are doing if you are root!

Reply:
> Those arguments will probably also apply to the password command, if it doesn't have such (and it doesn't seem like the password command has such options).

Oh my bad. When I talked about "borrowing ideas of `user password` command" I probably ment the `user modify` command - We have `--password` and `--password-prompt` there.

As we had it so often already with Click/synadm, copying boilerplate code is the quick way but it would be nice if we  could find a way to prepare these options once and reuse them in several commands. That also applies to any sanity-checking code we'd require on the UI end.

Reply:
Regarding reusing of options or even a couple of options, I think I found a nice way we could try. Creating a decorator that applies a group of options. Have a look at the very bottom of this post: https://stackoverflow.com/a/66743881/10207149

What do you think?

Reply:
> > * `--prompt` (default, mutually exclusive of `--no-prompt`)
> > * `--no-prompt` (mutually exclusive of `--prompt`, leave errors up to synapse)
> > * `--password` ([likely insecure](https://unix.stackexchange.com/questions/439497/is-there-a-way-to-pass-sensitive-data-in-bash-using-a-prompt-for-any-command) but useful for automation in testing environments, takes priority over stdin options)
> 
> Exactly, something like that!
> 
> > Those arguments will probably also apply to the password command, if it doesn't have such (and it doesn't seem like the password command has such options).
> 
> Oh my bad. When I talked about "borrowing ideas of `user password` command" I probably ment the `user modify` command - We have `--password` and `--password-prompt` there.
> 
> As we had it so often already with Click/synadm, copying boilerplate code is the quick way but it would be nice if we could find a way to prepare these options once and reuse them in several commands. That also applies to any sanity-checking code we'd require on the UI end.

Regarding password options streamlining, I would love if we could make use of the _common module I introduced here: https://github.com/JOJ0/synadm/pull/129/commits/79d0ea40eaf0e8101a3cca00cb88c73b1528a64d

We could put all the password related option into that module and reuse them with several password-related commands.

Reply:
Sure, certainly I ment once it's merged ;-)

Reply:
And yes: I agree we should also make use of that `click.password_option`

For some reason I did not use it in my intial implementation of `user password` - there might be a reason for it but it could also be that I just didn't read the docs carfully enough (it's been around since Click 6.x at least or even since always ;-). I used a confirmation prompt thing though :-)

```
$ synadm user password testuser3
Password:
Repeat for confirmation:
{}
```
https://github.com/JOJ0/synadm/blob/be77cef38e6889a37ec1b2acb9d28f2e58f83795/synadm/cli/user.py#L286-L288

We do also have use-cases were this option might not be enough....for example when handling with tokens there is more to it than either by prompt or by passing a cli arg... probably off-topic here....just mentioning..